oozie/src/main/java/org/apache/falcon/oozie/process/NativeOozieProcessWorkflowBuilder.java

```diff
@@ -0,0 +1,222 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.falcon.oozie.process;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.falcon.FalconException;
+import org.apache.falcon.entity.EntityUtil;
+import org.apache.falcon.entity.FeedHelper;
+import org.apache.falcon.entity.Storage;
+import org.apache.falcon.entity.store.ConfigurationStore;
+import org.apache.falcon.entity.v0.EntityType;
+import org.apache.falcon.entity.v0.cluster.Cluster;
+import org.apache.falcon.entity.v0.feed.Feed;
+import org.apache.falcon.entity.v0.feed.Location;
+import org.apache.falcon.entity.v0.feed.LocationType;
+import org.apache.falcon.entity.v0.process.Input;
+import org.apache.falcon.entity.v0.process.Output;
+import org.apache.falcon.entity.v0.process.Property;
+import org.apache.falcon.expression.ExpressionHelper;
+import org.apache.falcon.util.DateUtil;
+import org.apache.falcon.workflow.WorkflowExecutionArgs;
+import org.apache.hadoop.fs.Path;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Workflow Builder for oozie process in case of Native Scheduler.
+ */
+public class NativeOozieProcessWorkflowBuilder extends OozieProcessWorkflowBuilder {
+
+    private static final ExpressionHelper EXPRESSION_HELPER = ExpressionHelper.get();
+    private static final String INSTANCE_FORMAT = "yyyy-MM-dd-HH-mm";
+
+    public NativeOozieProcessWorkflowBuilder(org.apache.falcon.entity.v0.process.Process entity) {
+        super(entity);
+    }
+
+    @Override
+    public java.util.Properties build(org.apache.falcon.entity.v0.cluster.Cluster cluster,
+                                      Path buildPath, Properties suppliedProps) throws FalconException {
+        Properties elProps = new Properties();
+        DateTimeFormatter fmt = DateTimeFormat.forPattern(INSTANCE_FORMAT);
+        elProps.put(WorkflowExecutionArgs.NOMINAL_TIME.getName(), fmt.print(getNominalTime()));
+        elProps.put(WorkflowExecutionArgs.TIMESTAMP.getName(), fmt.print(getNominalTime()));
+        elProps.put(WorkflowExecutionArgs.USER_JMS_NOTIFICATION_ENABLED.getName(), "true");
+        elProps.put(WorkflowExecutionArgs.SYSTEM_JMS_NOTIFICATION_ENABLED.getName(), "false"); //check true or false
+
+
+        DateUtil.setTimeZone(entity.getTimezone().getID());
+        ExpressionHelper.setReferenceDate(new Date(getNominalTime().getMillis()));
+        elProps.putAll(getInputProps(cluster));
+        elProps.putAll(getOutputProps());
+        elProps.putAll(evalProperties());
+        Properties buildProps  = build(cluster, buildPath);
+        buildProps.putAll(elProps);
+        copyPropsWithoutOverride(buildProps, suppliedProps);
```

Comment:
I have taken this from existing one which is happening in oozie schedule , don't want to break backward compatibility.